test/stablex/batch_exchange.js

```diff
@@ -2063,11 +2083,17 @@ contract("BatchExchange", async accounts => {
         { from: solver }
       )
       assert(
-        firstSubmissionTX.receipt.gasUsed < sixPointFiveMillion,
-        `Solution submission exceeded 6.5 million gas at ${firstSubmissionTX.receipt.gasUsed}`
+        firstSubmissionTX.receipt.gasUsed < fivePointFiveMillion,
+        `Solution submission exceeded 5.5 million gas at ${firstSubmissionTX.receipt.gasUsed}`
       )
 
-      const solution2 = solutionSubmissionParams(longRingTrade.solutions[1], accounts, orderIds)
+      // Ensure second 30 order's users valid withdraw requests.
+      for (const deposit of tradeExample.deposits.slice(30)) {
```

Comment:
Doesn't this have to happen before the order collection is ended? Otherwise `hasValidWithdrawRequest` will be false and thus we won't hit [this line](https://github.com/gnosis/dex-contracts/blob/2dfb10ea610e12109460c499c548361c6beb7ce1/contracts/EpochTokenLocker.sol#L198) in the second solution submission.
Thus it's not the worst case.

Reply:
I don't understand the problem here. The orders have all been places and all of these assertions are evaluating to true.

Reply:
You are right. I though you were requesting it here, but you are just asserting.

<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia0.giphy.com%2Fmedia%2FYl5aO3gdVfsQ0%2Fgiphy.gif&token=nothing_to_see" data-gitmeme-token="nothing_to_see"><img src="https://media0.giphy.com/media/Yl5aO3gdVfsQ0/giphy.gif" title="Created by gitme.me with /nothing_to_see"/></a>